test/search/test_views.py

```diff
@@ -88,3 +84,29 @@ def test_catalogue_search_view(self):
             ],
         )
         self.assertEqual(response.context_data.get("bucket_keys"), BucketKeys)
+
+        # Assert for presence of the unchecked online checkbox where no group is set in request
+        response_no_group = self.client.get("/catalogue/search/")
+        html = force_str(response_no_group.content)
+        self.assertIn('name="online"', html)
+        self.assertNotIn('name="online" checked', html)
+
+        # Assert for checked state where there is no group and online is set to true in request
+        response_checked = self.client.get("/catalogue/search/?online=true")
+        html_checked = force_str(response_checked.content)
+        self.assertIn('name="online" checked', html_checked)
+
+        # Assert for presence of the unchecked online checkbox where group is set to 'tna'
+        response_group_tna = self.client.get("/catalogue/search/?group=tna")
+        html = force_str(response_group_tna.content)
+        self.assertIn('name="online"', html)
+        self.assertNotIn('name="online" checked', html)
+
+        # Assert for checked state where group is set to 'tna' and online is set to true in request
+        response_group_tna_online = self.client.get("/catalogue/search/?group=tna&online=true")
+        html_checked = force_str(response_group_tna_online.content)
+        self.assertIn('name="online" checked', html_checked)
+
+        # Assert the online checkbox is not included if group is set to 'nonTna'
+        non_tna_response = self.client.get("/catalogue/search/?group=nonTna")
+        self.assertNotIn('name="online"', force_str(non_tna_response.content))
```

Comment:
Hi  @ahosgood I want to make sure/confirm this please - When we moved from `ds-wagtail` to `ds-search`, a dev discussion was that we're not to writing tests to check for content in this repo as there will be other tests that check content. At this point, we're  checking for context vars that's rendered. 
 
 Perhaps when BE is integrated with the show online filter, there should be some tests for showing context results for `online=true`

Reply:
Hi @TNA-Allan - let me know what's decided here. I'm keen to follow any approach that's been agreed. 

My initial sense was that there is value in having tests to confirm the presence / absence of those filters that are bucket-specific (i.e. the digitised filter should only be available where the bucket is `tna`) but, if there's no value in these tests, I'm happy to remove them 👍🏽 . Let me know.

Reply:
@gtvj , lets keep this for now. 

Approving it. It can be removed if needed. 

Since it has been a long time we've come to this, maybe needs a discussion/clarity if we're going to have website tests, or content/integration tests in another folder and we can put in some guidance for development.

I think these tests could be split into another group for testing ` content/integration tests` and then update the current name `test_catalogue_search_view`-> `test_catalogue_search_view_context`. Alternatively put in a #TODO refactor this view for context and content tests - to come back it.